genericapi/handler.go

```diff
@@ -84,15 +92,28 @@ func (h *Handler) ServeCreateAlert(w http.ResponseWriter, r *http.Request) {
 
 	summary := r.FormValue("summary")
 	details := r.FormValue("details")
+	action := r.FormValue("action")
 
-	summary = validate.SanitizeText(summary, alert.MaxSummaryLength)
-	details = validate.SanitizeText(details, alert.MaxDetailsLength)
+	contentType := r.Header.Get("Content-type")
+	if contentType == "application/json" {
+		requestBoday, _ := ioutil.ReadAll(r.Body)
+
+		var alertMessage AlertMessage
+		json.Unmarshal(requestBoday, &alertMessage)
+
+		summary = alertMessage.Summary
+		details = alertMessage.Details
+		action = alertMessage.Action
```

Comment:
Now for these we can do a nil-check and ignore any missing values
```suggestion
		if b.Summary != nil {
			summary = *b.Summary
		}
		if b.Details != nil {
			details = *b.Details
		}
		if b.Action != nil {
			action = *b.Action
		}
```